File: Problems/Easy/1-TwoSum/Java/Solution2.java
package Java;

// import Java.util.HashMap;
// import Java.util.Map;

public class Solution2 {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> missingVal = new HashMap<>();
        int n = nums.length;
        for (int i =0; i<n;i++){
            int diff = target - nums[i];
            
            if (missingVal.containsKey(diff)){
                return new int[]{missingVal.get(diff),i};
            }

            missingVal.put(nums[i],i);
        }

            return new int[]{};
                
            }
      
    }
    /*
    * Runtime = 2 ms
    */

File: Problems/Easy/1-TwoSum/C++/Solution2.cpp
class Solution2 {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int, int> hashtable;
        for (int i = 0; i < n; ++i) {
            auto it = hashtable.find(target - nums[i]);
            if (it != hashtable.end()) {
                return {it->second, i};
            }
            hashtable[nums[i]] = i;
        }
        return {};
    }
};

// Runtime = 9ms